backend/src/db/providers/skillProvider.ts

```diff
@@ -194,6 +194,22 @@ const isProperTechnologies = async (techs: IUpdateCourseBody['skills']): Promise
   return checksPassed;
 };
 
+const skillsToCourseTechs = async (technologies: { skill: ObjectId; points: number }[]) => {
+  const techs = await Promise.all(
+    technologies.map(({ skill }) => {
+      return SkillModel.findOne({ name: skill });
```

Comment:
it's not quite clear for me how you search by name ( string in model) with { skill } that is equal to Objectid

Reply:
i consider that you can move this promise all out to controller and use instead this provider provider "getCommonSkill"